src/libcore/str/mod.rs

```diff
@@ -3554,6 +3554,8 @@ impl str {
     ///
     /// assert_eq!("Hello\tworld", s.trim());
     /// ```
+    #[must_use = "this returns the trimmed string as a new allocation, \
```

Comment:
I'm pretty sure `trim` [returns a slice](https://github.com/rust-lang/rust/blob/01f76d3db7a8419e250d29e4015678c4c08c444a/src/libcore/str/mod.rs#L3754) into the existing string and therefore doesn't allocate? (Sorry my [earlier comment](https://github.com/rust-lang/rust/pull/57106#discussion_r243861190) wasn't clearer: I linked to the `must_use` reason on `String.replace` just as an example of the reason functionality being used. Consistent language style is desirable, but we don't want to reuse exact words when that would be inaccurate.)